src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -57,8 +58,22 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
   }
 
   override def update(rows: Seq[Row]): IO[Unit] = {
-    val assetUpdates = rows.map(r => fromRow[AssetUpdate](r))
-    client.assets.update(assetUpdates) *> IO.unit
+    val assetUpdates = rows.map(r => fromRow[AssetsUpsertSchema](r))
+    val (updatesById, updatesByExternalId) = assetUpdates.partition(
+      p => p.id.isDefined
+    )
+    val id = if (updatesById.isEmpty) { IO.unit } else {
+      client.assets.updateById(updatesById.map(a => a.id.get -> a.transformInto[AssetUpdate]).toMap)
+    }
+    val extId = if (updatesByExternalId.isEmpty) { IO.unit } else {
+      client.assets.updateByExternalId(
+        updatesByExternalId
+          .map(a =>
+            a.externalId
+              .getOrElse("") -> a.into[AssetUpdate].withFieldComputed(_.externalId, _ => None).transform)
```

Comment:
I think this should be `.get`, actually: it should never happen due to the `.partition` above, and if it does happen we should crash (as `.get` does on `None`).

Reply:
my thought was that we partition on if id is non-null, so theoretically I guess externalId could be null here. In updateByExternalId() in the scala sdk, we require that the externalId is > "", so it ends up giving the error message "Updating by externalId requires externalId to be set".

Reply:
but if crashing here is more helpful, then can definitely change that easily